### PR TITLE
Planned coverage for VertX-specific metrics

### PR DESCRIPTION
**Pull request by fedinskiy**
### Links

JIRA: https://issues.redhat.com/browse/QUARKUS-2830

Quarkus documentation:
- https://quarkus.io/guides/vertx
- https://quarkus.io/guides/vertx-reference
- https://quarkus.io/guides/reactive-event-bus

### Reminder for considerable topics

 - [x] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-2830.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# QUARKUS-2829 - Full Support Micrometer metrics also when using managed Vert.x in Quarku
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-2830
+
+As part of supporting Vert.x use-cases on Quarkus we need to support collecting metrics using micrometer for the managed Vert.x instance in Quarkus. This is also a requirement for some layered products like AMQ Streams and OpenShift Serverless that today are using Vert.x, but needs to move to use Quarkus.
+
+## Scope  of the implementation
+Managed Vert.x in Quarkus contains following extensions:
+- Messaging clients:
+  AMQP,Kafka, MQTT. Extension names: quarkus-smallrye-reactive-messaging-amqp, *-kafka, *-mqtt.
```
Comment (rsvoboda) [inline QUARKUS-2830.md:10]:
```suggestion
  AMQP, Kafka, MQTT. Extension names: quarkus-smallrye-reactive-messaging-amqp, *-kafka, *-mqtt.
```

File: QUARKUS-2830.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Cover, that metrics are being created properly for applications, running on different targets (baremetal, openshift and serverless).
+
+### Existing tests
+Upstream tests contain coverage for Redis, HTTP client, telnet, eventbus, TCP and UDP metrics on baremetal.
+[Quarkus start-stop test suite](https://github.com/quarkus-qe/quarkus-startstop) contains several modules, which contain Vertx.X based extensions, the most interesting for our current efforts are monitoring-micrometer-prometheus-kafka-reactive which cover gathering of Micrometer metrics from AMQ Streams applications.
+As a part of QUARKUS-2829, we added checks for https server metrics on baremetal, openshift and serverless.
+
+## Automated test development
+We should add checks:
+- for databases properties, either in vertx-sql module, or add database usage into vertx-http module.
```
Comment (rsvoboda) [inline QUARKUS-2830.md:35]: I would prefer vertx-sql module, let's not polue vertx-http with DB stuff

File: QUARKUS-2830.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- UDP metrics (bytes_written, bytes_read, errors)
+- EventBus metrics (eventBus.published, eventBus.sent, replyFailures, handlers, delivered, discarded)
+- Http client metrics (queue.delay, queue.size)
+
+## Scope of testing
+Cover, that metrics are being created properly for applications, running on different targets (baremetal, openshift and serverless).
+
+### Existing tests
+Upstream tests contain coverage for Redis, HTTP client, telnet, eventbus, TCP and UDP metrics on baremetal.
+[Quarkus start-stop test suite](https://github.com/quarkus-qe/quarkus-startstop) contains several modules, which contain Vertx.X based extensions, the most interesting for our current efforts are monitoring-micrometer-prometheus-kafka-reactive which cover gathering of Micrometer metrics from AMQ Streams applications.
```
Comment (rsvoboda) [inline QUARKUS-2830.md:30]: Did you really mean StartStopTS? Or https://github.com/quarkus-qe/quarkus-test-suite?

File: QUARKUS-2830.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+As a part of QUARKUS-2829, we added checks for https server metrics on baremetal, openshift and serverless.
+
+## Automated test development
+We should add checks:
+- for databases properties, either in vertx-sql module, or add database usage into vertx-http module.
+- for event-bus specific properties on different targets
+
```
Comment (rsvoboda) [inline QUARKUS-2830.md:37]: Do you have plans to cover grpc metrics too? Where would be the coverage located?

Comment (rsvoboda) [inline QUARKUS-2830.md:37]: Another topic is impact on TS execution time, especially in native.

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-2830.md (hunk 3 above)
Comment (fedinskiy) [reply QUARKUS-2830.md]: I copied this from QUARKUS-2829. This doesn't seem to be relevant, so I dropped it.

File: QUARKUS-2830.md (hunk 4 above)
Comment (fedinskiy) [reply QUARKUS-2830.md:42]: Yes, that is a good idea. Added a plan for grpc.

**Review by rsvoboda — APPROVED** [review]
grpc coverage - https://github.com/quarkus-qe/quarkus-test-suite/blob/main/monitoring/opentelemetry-reactive looks like a better fit to me

**Pull request merged**